### PR TITLE
feat: added IFFT to stateful class for stateful layers

### LINKED ISSUE #22791: ifft

### PR DESCRIPTION
**Pull request by arshPratap**
# PR Description 

## Related Issue 

Close #22791 

## Checklist 

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by bipinKrishnan** [comment]
@arshPratap Can you please try rebasing on top of main, because I see some failures that are not directly related to your code.

**Comment by arshPratap** [comment]
Hey @bipinKrishnan ..I have rebased and ran the tests again..but it seems the tests are not running because of this error `ModuleNotFoundError: No module named ''`

File: ivy/stateful/layers.py [hunk 1]
```diff
@@ -2224,3 +2224,58 @@ def _forward(self, x):
...
+        dtype=None,
+    ):
+        """
+        Class for applying IFFT to input.
+
+        Parameters
+        ----------
+        dim : int
+            Dimension along which to take the IFFT.
+        norm : str
```
Comment (bipinKrishnan) [inline ivy/stateful/layers.py:2248]: May be we could add the possible values for this argument as in [here](https://github.com/unifyai/ivy/blob/1865ed37f143f731913fcb6ac03fe23fb90c989e/ivy/data_classes/array/experimental/layers.py#L612)

File: ivy/stateful/layers.py [hunk 2]
```diff
@@ -2224,3 +2224,58 @@ def _forward(self, x):
...
+            Normalization mode. Default: 'backward'
+        n : int
```
Comment (bipinKrishnan) [inline ivy/stateful/layers.py:2250]: This details of this argument could be expanded by referring [here](https://github.com/unifyai/ivy/blob/1865ed37f143f731913fcb6ac03fe23fb90c989e/ivy/data_classes/array/experimental/layers.py#L618).

File: ivy/stateful/layers.py [hunk 3]
```diff
@@ -2224,3 +2224,58 @@ def _forward(self, x):
...
+        Forward pass of the layer.
+
+        Parameters
+        ----------
+        inputs : array
+            Input array to take the IFFT of.
+
+        Returns
+        -------
+        array
```
Comment (bipinKrishnan) [inline ivy/stateful/layers.py:2272]: There is no variable named `array` that is returned, so you could remove this line.

**Comment by bipinKrishnan** [comment]
We could merge the PR once the suggestions are incorporated, the tests are failing due to some issue not related your changes.

File: ivy/stateful/layers.py (hunk 3 above)
Comment (arshPratap) [reply ivy/stateful/layers.py]: thanks @bipinKrishnan ..have made the changes

File: ivy/stateful/layers.py (hunk 2 above)
Comment (arshPratap) [reply ivy/stateful/layers.py:2254]: Thanks @bipinKrishnan ..have made the changes

File: ivy/stateful/layers.py (hunk 1 above)
Comment (arshPratap) [reply ivy/stateful/layers.py:2248]: Thanks @bipinKrishnan ..have made the changes

**Comment by arshPratap** [comment]
Hi @bipinKrishnan ..Could you take a look. Have rebased to the current `main` and have made the requested changes

**Comment by arshPratap** [comment]
> We could merge the PR once the suggestions are incorporated, the tests are failing due to an issue not related your changes.

@bipinKrishnan Done ! Could you take another look .. Thanks

**Comment by bipinKrishnan** [comment]
LGTM! Let's wait for the tests to complete.

**Comment by arshPratap** [comment]
Hi @bipinKrishnan ..no `failing tests` are introduced..looks good..Could you take another look !

**Comment by bipinKrishnan** [comment]
Thanks for the contribution @arshPratap 🚀 !

**Pull request merged**